src/main/java/gov/usds/case_issues/config/OAuth2ClientRegistrationCondition.java

```diff
@@ -0,0 +1,28 @@
+package gov.usds.case_issues.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientProperties;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * Condition implementation to enable definition of beans that are only registered
+ * when there is an OAuth2 client registration in the configuration.
+ */
+public class OAuth2ClientRegistrationCondition extends SpringBootCondition {
```

Comment:
There is an argument for just inlining this into the configuration class that is conditional on it, since there's no short-term prospect of anything else using it anyway.